server/aap/commands/fulfill_image_assignments.py

```diff
@@ -0,0 +1,225 @@
+# AUTHORS and LICENSE files distributed with this source code, or
+# at https://www.sourcefabric.org/superdesk/license
+
+import logging
+import superdesk
+from superdesk.utc import utcnow, utc_to_local
+from datetime import timedelta
+from planning.common import ASSIGNMENT_WORKFLOW_STATE
+from eve.utils import ParsedRequest
+from flask import json
+from flask import current_app as app
+import requests
+from lxml import etree
+from superdesk.celery_task_utils import get_lock_id
+from superdesk.lock import lock, unlock
+
+logger = logging.getLogger(__name__)
+
+
+class FullfillImageAssignments(superdesk.Command):
+    # The headers returned by the login request to the DC Rest service
+    dc_headers = dict()
+
+    def _dc_login(self):
+        """
+        Function will log into the DC system rest API and save the response headers
+        :return:
+        """
+        url = app.config.get('DC_URL')
+        username = app.config.get('DC_USERNAME')
+        password = app.config.get('DC_PASSWORD')
+
+        response = requests.get(url + '/?login[username]={}&login[password]={}'.format(username, password))
+        response.raise_for_status()
+
+        self.dc_headers = response.headers
+
+    def _get_dc_image_by_id(self, collection, id):
+        """
+        This function given a DC collection and the DC id of an image will return the meta data from DC for that image
+        :param collection:
+        :param id:
+        :return: The XML response that should contain the metadata for the single image
+        """
+        url = app.config.get('DC_URL')
+
+        retires = 0
+        while retires < 3:
+            try:
+                response = requests.get(url + '/archives/{}?search_docs[query]=id%3d'
+                                              '{}&search_docs[format]=full'.format(collection, id),
+                                        headers={'cookie': self.dc_headers.get('Set-Cookie')})
+                response.raise_for_status()
+                break
+            except requests.exceptions.RequestException:
+                try:
+                    self._dc_login()
+                except requests.exceptions.RequestException:
+                    logger.error('DC login failed')
+                retires += 1
+        if retires == 3:
+            logger.error('Failed to get image assignment from DC {}'.format(id))
+            return None
+        return etree.fromstring(response.content)
+
+    def _get_dc_images_by_transref(self, archive, transref):
+        url = app.config.get('DC_URL')
+
+        here_now = utc_to_local(app.config['DEFAULT_TIMEZONE'], utcnow())
+        start = (here_now - timedelta(days=2)).strftime('%Y%m%d')
+
+        retries = 0
+        while retries < 3:
+            try:
+                response = requests.get(url + '/archives/{}?search_docs[query]='
+                                              '(ORIGINALTRANSMISSIONREFERENCE%3D{})%26(MODDATE%3E{})'
```

Comment:
Should we also have a setting for the lookup attribute from DC? Similar to how `PLANNING_XMP_ASSIGNMENT_MAPPING` is used when setting the XMP attribute?

Reply:
Done and tested with keywords!